src/CarbonAware.WebApi/src/Filters/HttpResponseExceptionFilter.cs

```diff
@@ -9,62 +9,75 @@ namespace CarbonAware.WebApi.Filters;
 public class HttpResponseExceptionFilter : IExceptionFilter
 {
     private ILogger<HttpResponseExceptionFilter> _logger;
-    private static readonly ActivitySource _activitySource = new ActivitySource(nameof(HttpResponseExceptionFilter));
+    private IConfiguration _config;
 
     private static Dictionary<string, int> EXCEPTION_STATUS_CODE_MAP = new Dictionary<string, int>()
     {
         { "ArgumentException", (int)HttpStatusCode.BadRequest },
         { "NotImplementedException", (int)HttpStatusCode.NotImplemented },
     };
 
-    public HttpResponseExceptionFilter(ILogger<HttpResponseExceptionFilter> logger)
+    public HttpResponseExceptionFilter(ILogger<HttpResponseExceptionFilter> logger, IConfiguration configuration)
     {
         _logger = logger;
+        _config = configuration;
     }
 
     public void OnException(ExceptionContext context)
     {
-        using (var activity = _activitySource.StartActivity())
-        {
+        var activity = Activity.Current;
```

Comment:
Without a "startActivity()" call, distributed tracing will be broken.  This usage is incorrect.

Reply:
Hi @rakker91 . We ran into this and the activities are `null` all the time, since there isn't any listener. The `workaround` was to put it this way to get back a `traceId`, but still there is a bigger question now on how all these calls to `ActivitySource` are done. Mainly from the perspective of where and how and what would be the overall story around it. Seems like we are missing that.

Reply:
Yes, we noticed that it wasnt setup currently. We are planning to investigate more into this and address this as part of a new story.